authentication/ruby/oauth/microsoft_kiota_authentication_oauth/lib/microsoft_kiota_authentication_oauth/contexts/oauth_context.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'oauth2'
+
+module MicrosoftKiotaAuthenticationOAuth
+    # Base class token request context and can be optionally implemented for 
+    # custom token grant flows.
+    class OAuthContext
```

Comment:
```suggestion
    module OAuthContext
```

Reply:
In Ruby, modules are collections of functions, so if I change it to a module, it can't be inherited by anything, and it can't have objects/instances or a constructor, so I left it as a class, but maybe I should add a comment saying that our supported flows are subclasess of OAuthContext to remedy this?

Reply:
but modules can be included in each context? (from looking at how abstractions have been implemented)
If so, I may have misled you on `is_a`, not sure what the appropriate method to check whether a module is included is.

Reply:
Also don't think `initialize` should be part of the OAuthContext since it's a constructor

Reply:
@baywet composition over inheritance? Pending discussion is around the base class vs interface approach using modules

Reply:
we had a quick chat about that with Osose yesterday. As far as I understand ruby interfaces are not a thing, you have to use modules. But modules can declare fields + corresponding accessors, only classes can. Think of that as a base class that defines both a contract and some very basic behavior (fields that will be need by implementers anyway). I hope that makes sense?

Reply:
Was thinking we don't need to declare any fields but instead define initialize_scopes(scopes), get_token and get_oauth_provider as our contract

& removing initialize and the instance variables.

Reply:
I'm ok either way, my only thought was it'd save some duplicated work to implementers as they'll need some of those fields no matter what.

Reply:
Ah ok. Then I think it's fine since interfaces are not really  a Ruby-native thing.